utils_cv/detection/dataset.py

```diff
@@ -218,20 +208,33 @@ def _read_annos(self) -> List[str]:
                 os.path.splitext(s)[0] + ".xml" for s in im_filenames
             ]
 
-        # Parse all annotations
+        # Read all annotations
         self.im_paths = []
         self.anno_paths = []
         self.anno_bboxes = []
         for anno_idx, anno_filename in enumerate(anno_filenames):
             anno_path = self.root / self.anno_dir / str(anno_filename)
-            assert os.path.exists(
-                anno_path
-            ), f"Cannot find annotation file: {anno_path}"
-            anno_bboxes, im_path = parse_pascal_voc_anno(anno_path)
 
-            # TODO For now, ignore all images without a single bounding box in it, otherwise throws error during training.
+            # Parse annotation file if present
+            if os.path.exists(anno_path):
+                anno_bboxes, im_path = parse_pascal_voc_anno(anno_path)
+            else:
+                if self.require_annotation_files:
+                    raise FileNotFoundError(anno_path)
+                anno_bboxes = [] 
+                im_path = im_paths[anno_idx]
+
+            # Torchvision needs at least one ground truth bounding box per image. Hence for images without a single
+            # annotated object, adding a tiny bounding box with "background" label 0.
             if len(anno_bboxes) == 0:
-                continue
+                anno_bboxes = [
+                    AnnotationBbox.from_array(
+                        [1, 1, 5, 5],
+                        label_name=None,
```

Comment:
Note the special case here: I am setting the label_name to None since this label does not correspond to any of the objects-of-interest. Alternative would be to set it to "__background__", but that would also require some code edits so that nothing breaks